src/AutoRest.CSharp/LowLevel/Output/OperationMethodChainBuilder.cs

```diff
@@ -98,7 +99,8 @@ public LowLevelClientMethod BuildOperationMethodChain()
         private bool ShouldConvenienceMethodGenerated(ReturnTypeChain returnTypeChain)
         {
             return Operation.GenerateConvenienceMethod
-                && (_orderedParameters.Where(parameter => parameter.Convenience != KnownParameters.CancellationTokenParameter).Any(parameter => !IsParameterTypeSame(parameter.Convenience, parameter.Protocol))
+                && (!Operation.GenerateProtocolMethod
```

Comment:
Why do we add this check here?  An operation must generate one method, protocol, convenience or both.  And at least one of the flag ( GenerateConvenienceMethod, GenerateProtocolMethod) is set true.